src/nih_wayfinding/app/scripts/routing/abbreviate-filter.js

```diff
@@ -0,0 +1,105 @@
+
+(function () {
+    'use strict';
+
+    var directions = {
+        north: 'N',
+        northeast: 'NE',
+        east: 'E',
+        southeast: 'SE',
+        south: 'S',
+        southwest: 'SW',
+        west: 'W',
+        northwest: 'NW',
+        n: 'N',
```

Comment:
It's a side effect of the way the linked python solution did it. You check the keys dict, but when you lowercase to get the keys, 'n' no longer matches 'N'. I like that doing it this way keeps additional complexity out of the if block logic in the function body, but I can play with a cleaner solution.